corehq/apps/api/resources/v0_5.py

```diff
@@ -1122,56 +1191,30 @@ def dehydrate(self, bundle):
 
     def obj_get_list(self, bundle, **kwargs):
         domain = kwargs['domain']
-        params = self._process_params(domain, bundle.request.GET)
+        self.api_params = self._process_params(domain, bundle.request.GET)
 
-        results = self.cursor_query(domain, self.local_timezone, params)
+        results = self.cursor_query(domain, self.api_params)
         return list(map(self.to_obj, results))
 
     def _process_params(self, domain, params):
-        processed_params = {}
+        api_params = NavigationEventAuditResourceParams(raw_params=params)
 
-        for param in params:
-            val = params.get(param)
-            if param == 'users':
-                val = params.getlist(param)
-            processed_params[param] = val
+        if api_params.limit:
+            api_params.process_limit(self._meta.limit, self._meta.max_limit)
+        if api_params.cursor:
+            api_params.process_cursor()
+        api_params.process_local_timezone(domain)
```

Comment:
What would you think about moving these calls to `__post_init__`?

Aside: an alternative to `dataclasses` - `attrs` - provides validator and converter hooks that would be useful here: https://www.attrs.org/en/stable/init.html#validators
(though I think this is fine, it's not necessary to switch libraries)